samcli/lib/providers/sam_function_provider.py

```diff
@@ -387,3 +388,13 @@ def _parse_layer_info(
                 )
 
         return layers
+
+    def get_resources_by_stack_path(self, stack_path: str) -> Dict:
+        candidates = [
+            resources
+            for stack, resources in self.stack_and_resources
+            if stack.stack_path_for_children_resources == stack_path
+        ]
+        if not candidates:
+            raise RuntimeError(f"Cannot find resources with stack_path = {stack_path}")
+        return candidates[0]
```

Comment:
Would `stack_path` be unique amongst the resources in all stacks? If so should we check if there are more than 1 candidate here?

Reply:
> Would stack_path be unique amongst the resources in all stacks?

Yes, the `resources` means "all resources (func/layer) in a single stack," so it has a unique stack_path